lib/discourse_hub.rb

```diff
@@ -52,14 +52,25 @@ def self.singular_action(action, rel_url, params = {})
 
   def self.collection_action(action, rel_url, params = {})
     connect_opts = connect_opts(params)
-    JSON.parse(Excon.send(action,
+
+    response = Excon.send(action,
       "#{hub_base_url}#{rel_url}",
       {
         body: JSON[params],
         headers: { 'Referer' => referer, 'Accept' => accepts.join(', '), "Content-Type" => "application/json" },
         omit_default_port: true
       }.merge(connect_opts)
-    ).body)
+    )
+
+    if response.status != 200
+      log(:warninig, "Discourse Hub (#{hub_base_url}#{rel_url}) returned a bad status #{response.status}.")
+    end
+
+    begin
+      JSON.parse(response.body)
+    rescue
```

Comment:
When rescuing errors, we want to avoid rescuing all types of errors because errors like a typo will be incorrectly rescued as well. In this case, we should only rescue `JSON::ParserError`.